Atomic/info.py

```diff
@@ -104,123 +98,32 @@ def info(self):
-                    return buf
-            else:
-                self._no_such_image()
-
-        elif self.args.storage.lower() == "docker":
-            if self.is_iid():
-                self.get_fq_name(self._inspect_image())
-            # The input is not an image id
-            else:
-                try:
-                    iid = self._is_image(self.image)
-                    self.image = self.get_fq_name(self._inspect_image(iid))
-                except AtomicError:
-                    if self.args.force:
-                        self.image = util.find_remote_image(self.d, self.image)
-                    if self.image is None:
-                        self._no_such_image()
 
-        else:
-            raise ValueError("{} is not a valid storage".format(self.args.storage))
-
-        buf += ("Image Name: {}".format(self.image))
-        inspection = None
-        if not self.args.force:
-            inspection = self._inspect_image(self.image)
-            # No such image locally, but fall back to remote
-        if inspection is None:
-            # Shut up pylint in case we're on a machine with upstream
-            # docker-py, which lacks the remote keyword arg.
-            # pylint: disable=unexpected-keyword-arg
-            inspection = util.skopeo_inspect("docker://" + self.image)
-            # image does not exist on any configured registry
-        if 'Config' in inspection and 'Labels' in inspection['Config']:
-            labels = inspection['Config']['Labels']
-        elif 'Labels' in inspection:
-            labels = inspection['Labels']
-        else:
-            _no_label()
+        if self.args.storage == 'ostree' and self.args.force:
+            # Ostree and remote combos are illegal
+            raise ValueError("The --remote option cannot be used with the 'ostree' storage option.")
 
-        if labels is not None and len(labels) is not 0:
-            for label in labels:
-                buf += ('\n{0}: {1}'.format(label, labels[label]))
+        if self.args.force:
+            # The user wants information on a remote image
+            be = self.beu.get_backend_from_string(self.args.storage)
+            img_obj = be.make_remote_image(self.image)
         else:
-            _no_label()
-        return buf
+            # The image is local
+            be, img_obj = self.beu.get_backend_and_image(self.image, str_preferred_backend=self.args.storage)
+
+        with closing(StringIO()) as buf:
+            try:
+                info_name = img_obj.fq_name
+            except RegistryInspectError:
+                info_name = img_obj.input_name
+            buf.write("Image Name: {}\n".format(info_name))
+            buf.writelines(sorted(["{}: {}\n".format(k, v) for k,v in list(img_obj.labels.items())]))
+            if img_obj.template_variables_set:
+                buf.write("\n\nEnvironment variables with default value, but overridable with --set:\n")
```

Comment:
let's use "Template variables", as they are not passed as env variables for system containers.